File: OCH_Tao/ABC/B/265.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
let [N, , T] = input.shift().split(" ").map(x => parseInt(x, 10));
const A = input.shift().split(" ").map(x => parseInt(x, 10));
let D = new Map();
for (const i of input) {
  const [X, Y] = i.split(" ").map(x => parseInt(x, 10));
  D.set(X, Y);
}
for (let i = 1; i < N; i++) {
  T += (D.get(i) ?? 0);
  T -= A[i - 1];
  if (T <= 0) {
    console.log("No");
    process.exit();
  }
}
console.log("Yes");

File: OCH_Tao/ABC/B/265.py
N,M,T = map(int,input().split())
A = list(map(int,input().split()))
D = dict()
for i in range(M):
  X,Y = map(int,input().split())
  D[X] = Y
for i in range(1,N):
  T += D.get(i,0)
  T -= A[i-1]
  if T<=0:
    print("No")
    break
else:
  print("Yes")